tests/conftest.py

```diff
@@ -2598,3 +2598,152 @@ def ceph_toolbox(request):
+            check_ns_resource_validity(mcg_obj, rand_ns_resource,
+                                       target_bucket_name, constants.mcg_ns_aws_endpoint)
+
+            created_ns_resources.append(rand_ns_resource)
+            created_ns_connections.append(rand_connection)
+        return target_bucket_name, rand_ns_resource
+
+    def ns_resources_and_connections_cleanup():
+        for ns_resource in created_ns_resources:
+            delete_ns_resource(ns_resource, mcg_obj)
+
+        for ns_connection in created_ns_connections:
+            delete_ns_connection(ns_connection, mcg_obj)
+
+    request.addfinalizer(ns_resources_and_connections_cleanup)
+
+    return _create_ns_resources
+
+
+@pytest.fixture()
+def ns_bucket_factory(request, mcg_obj, cld_mgr):
+    """
+    Create a namespace bucket factory. Calling this fixture creates a new namespace bucket(s).
+    For a custom amount, provide the 'amount' parameter.
+
+    Args:
+        mcg_obj (MCG): An MCG object
+        cld_mgr (CM): A cloud manager object containing the MCG S3 connection
+        credentials
+
+    """
+    created_ns_buckets = []
+
+    def _create_ns_buckets(amount=1, write_resource=None, read_resources=None, *args, **kwargs):
+        if write_resource is None:
+            write_resource = []
+        if read_resources is None:
+            read_resources = [[]]
+        for i in range(amount):
+            # Create random namespace bucket name
+            rand_ns_bucket = create_unique_resource_name('aws', constants.mcg_ns_bucket)
+
+            # Create actual namespace bucket
+            mcg_obj.create_new_namespace_bucket(rand_ns_bucket, write_resource[i], read_resources[i])
+            created_ns_buckets.append(rand_ns_bucket)
+            check_ns_bucket_validity(mcg_obj, rand_ns_bucket, write_resource[i], read_resources[i])
+        return rand_ns_bucket
+
+    def ns_buckets_cleanup():
+        for ns_bucket in created_ns_buckets:
+            delete_ns_bucket(ns_bucket, mcg_obj)
+
+    request.addfinalizer(ns_buckets_cleanup)
+
+    return _create_ns_buckets
+
+
+def check_ns_resource_validity(mcg_obj, ns_resource_name, target_bucket_name, endpoint):
+    """
+    Check namespace resource validity
+    """
+    # Retrieve the NooBaa system information
+    system_state = mcg_obj.send_rpc_query('system_api', 'read_system').json().get('reply')
+
+    # Retrieve the correct namespace resource info
+    match_resource = [
+        ns_resource for ns_resource in system_state
+        .get('namespace_resources') if ns_resource.get('name') == ns_resource_name
+    ]
+    assert match_resource, f"The NS resource named {ns_resource_name} was not found"
+    actual_target_bucket = match_resource[0].get('target_bucket')
+    actual_endpoint = match_resource[0].get('endpoint')
+
+    assert actual_target_bucket == target_bucket_name, f"The NS resource named {ns_resource_name} " \
+        f"got wrong target bucket {actual_target_bucket} ≠ {target_bucket_name}"
```

Comment:
Please follow https://github.com/red-hat-storage/ocs-ci/blob/master/docs/coding_guidelines.md and use paranthase to break lines:
```suggestion
    assert actual_target_bucket == target_bucket_name, (
        f"The NS resource named {ns_resource_name} got "
        f"wrong target bucket {actual_target_bucket} ≠ {target_bucket_name}"
    )
```